src/axom/inlet/examples/document_generation_example.cpp

```diff
@@ -0,0 +1,173 @@
+
+  SLIC_ASSERT_MSG(sidreGroup->hasView("foo/description"), "foo/description not found");
+  SLIC_ASSERT_MSG(sidreGroup->hasView("bar/description"), "bar/description not found");
+  SLIC_ASSERT_MSG(sidreGroup->hasView("Table1/float1/description"), "Table1/float1/description not found");
+  SLIC_ASSERT_MSG(sidreGroup->hasView("Table1/Table11/Table111/x/description"), "Table1/Table11/Table111/x/description not found");
+}
+
+void createInletEx3(std::shared_ptr<Inlet> inlet)
+{
+  auto t = inlet->addTable("Table1", "The first table");
+  t->addDouble("float1", " A floating point number in Table 1");
+  t = inlet->addTable("Table2", "The second table");
+  t->addInt("int1", "An integer in Table 2");
+  t = inlet->addTable("Table3", "The third table");
+  t->addBool("bool1", "A boolean value in Table 3");
+
+  axom::sidre::Group* sidreGroup = inlet->sidreGroup();
+
+  bool found;
+  bool boolVal;
+  int intVal;
+  double doubleVal;  
+
+  found = inlet->get("Table1/float1", doubleVal);
+  SLIC_ASSERT_MSG(found == true, "Table1/float1 not found");
+  found = inlet->get("Table2/int1", intVal);
+  SLIC_ASSERT_MSG(found == true, "Table2/int1 not found");
+  found = inlet->get("Table3/bool1", boolVal);
+  SLIC_ASSERT_MSG(found == true, "Table3/bool1 not found");
+
+  SLIC_ASSERT_MSG(sidreGroup->hasView("Table1/float1/description"),"Table1/float1/description not found");
+  SLIC_ASSERT_MSG(sidreGroup->hasView("Table2/int1/description"), "Table2/int1/description not found");
+  SLIC_ASSERT_MSG(sidreGroup->hasView("Table3/bool1/description"), "Table3/bool1/description not found");
+}
+
+std::shared_ptr<Inlet> createExampleInlet(DataStore* ds,
+                                        const std::string& luaFile, int exNum)
+{
+  auto lr = std::make_shared<LuaReader>();
+  // lr->parseString(luaString);
+  lr->parseFile(luaFile);
+  auto inlet = std::make_shared<Inlet>(lr, ds->getRoot());
+  if (exNum == 1) {
+    createInletEx1(inlet);
+  } else if (exNum == 2) {
+    createInletEx2(inlet);
+  } else {
+    createInletEx3(inlet);
+  }
+  return inlet;
+}
+
+int main(int argc, char** argv) {
+  CLI::App app {"Description here"};
+  bool docs_enabled{false};
+  app.add_flag("--enableDocs", docs_enabled, "Enables documentation generation");
+
+  std::string strOption;
+  auto opt = app.add_option("--deck", strOption, "Path to input deck file");
+  opt->check(CLI::ExistingFile);
+
+  int exampleNum{1};
+  app.add_option("--example", exampleNum, "Example number to be run");
+
+  bool isVerbose{false};
+  app.add_flag("--verbose", isVerbose, "Enables output of inlet information during run-time");
+
+  CLI11_PARSE(app, argc, argv);
+
+  DataStore ds;
+  std::shared_ptr<Inlet> inlet = createExampleInlet(&ds, strOption, exampleNum);
+
+  std::cout << "____________________________________________" << std::endl;
+  if (docs_enabled) {
+    DocWriter doc("example_docs.rst", inlet->sidreGroup(), isVerbose);

```

Comment:
Can you change this to to a slightly different user experience.  I was thinking more along the lines of:

// Add doc mode to inlets creation so we can toggle saving the descriptions
std::shared_ptr<Inlet> inlet = std::make_shared<Inlet>(lr, sidre->getRoot(), docs_enabled);

...

// Create and register the docWriter with inlet, the constructor shouldn't write the docs
std::shared_ptr<DocWriter> docWriter = std::make_shared<DocWriter>("examples_docs.rst", isVerbose);
inlet->registerDocWriter(docWriter);

// Add new function to the Inlet class that calls the docWriter if documenation mode is turned on
inlet->writeDocs();